src/tokens/ERC1155/presets/lootbox/ERC1155Lootbox.sol

```diff
@@ -0,0 +1,114 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.19;
+
+import {IERC1155Lootbox, IERC1155LootboxFunctions} from "./IERC1155Lootbox.sol";
+import {ERC1155Items} from "@0xsequence/contracts-library/tokens/ERC1155/presets/items/ERC1155Items.sol";
+import {MerkleProof} from "@openzeppelin/contracts/utils/cryptography/MerkleProof.sol";
+import {IERC1155ItemsFunctions} from "@0xsequence/contracts-library/tokens/ERC1155/presets/items/IERC1155Items.sol";
+
+contract ERC1155Lootbox is ERC1155Items, IERC1155Lootbox {
+    bytes32 internal constant MINT_ADMIN_ROLE = keccak256("MINT_ADMIN_ROLE");
+
+    bytes32 public merkleRoot;
+    uint256 public boxSupply;
+
+    mapping(address => uint256) private _commitments;
+    mapping(uint256 => bool) private _claimedIdxs;
+
+    constructor() ERC1155Items() {}
+
+    /// @inheritdoc ERC1155Items
+    function initialize(
+        address owner,
+        string memory tokenName,
+        string memory tokenBaseURI,
+        string memory tokenContractURI,
+        address royaltyReceiver,
+        uint96 royaltyFeeNumerator
+    ) public virtual override {
+        _grantRole(MINT_ADMIN_ROLE, owner);
+        super.initialize(owner, tokenName, tokenBaseURI, tokenContractURI, royaltyReceiver, royaltyFeeNumerator);
+    }
+
+    /// @inheritdoc IERC1155LootboxFunctions
+    function setBoxContent(bytes32 _merkleRoot, uint256 _boxSupply) external onlyRole(MINT_ADMIN_ROLE) {
+        merkleRoot = _merkleRoot;
+        boxSupply = _boxSupply;
+    }
+
+    /// @inheritdoc IERC1155LootboxFunctions
+    function commit() external {
```

Comment:
Can we handle on1155Receive 
Methods to call commit()? This way users don't need to do an approval first to commit N loots

Reply:
Currently is not necessary to approve the NFTs before commit, as commit() directly calls the _burn function. This way we are not receiving and transferring back in case of expired commit, just burn/mint